### DISCUSSION TITLE
Is the profiling API available on Alpine Arm64 docker images?

### DISCUSSION THREAD

**Discussion by nrcventura** [dotnet/runtime#87519] [Q&A]
I tested out a few different images with a simple ICorProfiler that just logs a message to the console if it gets initialized (and the New Relic profiler), and it seems like the ICorProfiler is not getting invoked for `6.0-alpine-arm64v8` images. The same profiler is working as expected for `6.0-bullseye-slim-arm64v8` images.

Is there a way to confirm that the ICorProfiler profiling API is not available in that distribution? In general it would be good to know which distributions do not support the ICorProfiler profiling API.

**Comment by jkotas** [comment]
ICorProfiler profiling API is available in all distributions.

Alpine is musl-based distro. The profiler binary for glibc-based distro (e.g. Debian or Ubuntu) is not going to work on Alpine. You need to rebuild your profiler for Alpine (or some other musl-based distro) to get a profiler binary that works on Alpine.

**Comment by nrcventura** [comment]
This same profiler code is working on Alpine x64, so that's why I'm surprised that this same code is not working on Alpine Arm64.

**Comment by nrcventura** [comment]
The behavior I'm seeing reminds me of the behavior that I previously saw with windows nano server, where I didn't see any attempts to load the profiler which I tried to confirm by searching through the event viewer logs (back in 2018).

**Comment by nrcventura** [comment]
With the advice from @davmason we were able to use `ldd` to identify why the profiler wasn't getting loaded. The way we built that profiler required an additional dependency to be installed in that image, and once that dependency was available the profiler got loaded.
